api/nakadi-event-bus-api.yaml

```diff
@@ -0,0 +1,1232 @@
+
+      A basic payload of an Event. The actual schema is dependent on the information configured for the
+      EventType, as is its enforcement (see POST /event-types). Setting of metadata properties are
+      dependent on the configured enrichment as well.
+
+      For explanation on default configurations of validation and enrichment, see documentatin of
+      `EventType.type`.
+
+      For concrete examples of what will be enforced by Nakadi see the objects BusinessEvent and
+      DataChangeEvent below.
+    additionalProperties: true
+
+
+  EventMetadata:
+    type: object
+    description: |
+      Metadata for this Event.
+
+      Contains commons fields for both Business and DataChange Events. Most are enriched by Nakadi upon
+      reception, but they in general MIGHT be set by the client.
+    properties:
+      metadata:
+        type: object
+        properties:
+          eid:
+            description: |
+              Identifier of this Event.
+
+              Clients are allowed to generate this and this SHOULD be guaranteed to be unique from the
+              perspective of the producer. Consumers MIGHT use this value to assert uniqueness of reception
+              of the Event.
+
+              If not present upon reception, Nakadi will generate and fill this with a unique value.
+              (TBD: enforce UUID formatif set by the client?)
+            type: string
+            example: '105a76d8-db49-4144-ace7-e683e8f4ba46'
+          event_type:
+            description: |
+              The EventType of this Event. This is enriched by Nakadi on reception of the Event based on the
+              endpoint where the Producer sent the Event to.
+
+              If provided MUST match the endpoint. Failure to do so will cause rejection of the Event.
+            type: string
+            example: 'pennybags:payment-business-event'
+          occurred_at:
+            description: |
+              Timestamp of creation of the Event generated by the producer. Presence is optional but strongly
+              recommended for producers.
+            type: string
+            format: date-time
+          received_at:
+            type: string
+            description: |
+              Timestamp of the reception of the Event by Nakadi. This is enriched upon reception of the
+              Event.
+              If set by the producer Event will be rejected.
+            format: date-time
+          parent_eid:
+            type: string
+            format: uuid
+            description: |
+              Event identifier of the Event that caused the generation of this Event.
+              Set by the procuder.
+            example: '105a76d8-db49-4144-ace7-e683e8f4ba46'
+          root_eid:
+            type: string
+            format: uuid
+            description: |
+              Event identifier of the root of the tree of Events (from the relation constructed from
+              parent-eid)
```

Comment:
The complexity of usage is directly bound to the complexity of the usage of the `parent-eid`. 

Also `root-eid` has the same semantics as `flow-id` any, perhaps, having both these numbers does not make sense, because the only reason of having a `root-eid` is to simplify the lookup of the root of the chain of events.